README.md
# Geotrigger SDK for Android

The Geotrigger Service provides real-time location tracking, geofencing,
and messaging tools for developers working on location based apps.

The Geotrigger SDK provides the `GeotriggerService`, a wrapper around 
Android's native location capabilities, which handles uploading device locations
to the Geotrigger API and receiving push notifications.

See the [Getting Started with the Geotrigger SDK for Android][geotrigger-docs-android-getting-started] guide
for more information about the Geotrigger SDK.

Guides for other topics are available as part of the [Geotrigger Documentation][geotrigger-docs],
including interacting with the `GeotriggerService`, working with the Geotrigger API, and configuring
push notifications. Information about other platforms and the API itself are also available.

## Setup

The Geotrigger Service can be included in a project by modifying the following Gradle files.

Top-level `build.gradle`:

```groovy
allprojects {
    repositories {
        ...
        // Repository for the Esri Geotrigger SDK for Android
        maven {
            url 'https://esri.bintray.com/android'
        }
    }
}
```

App `build.gradle`:

```groovy
dependencies {
    ...
    // Esri Geotrigger SDK for Android
    compile 'com.esri.android.geotrigger:geotrigger-sdk:1.2.0'
}
```

[esri-site]: http://www.esri.com
[arcgis-dev-site]: https://developers.arcgis.com/
[geotrigger-docs]: https://developers.arcgis.com/en/geotrigger-service
[geotrigger-docs-android-getting-started]: https://developers.arcgis.com/en/geotrigger-service/guide/android-getting-started
[geotrigger-docs-android-push-notifications]: https://developers.arcgis.com/en/geotrigger-service/guide/android-push-notifications/
[android-docs]: http://developer.android.com/
[android-sdk]: http://developer.android.com/sdk/index.html
[google-play-services-setup]: http://developer.android.com/google/play-services/setup.html
[sample-app-geotrigger-activity]: https://github.com/Esri/geotrigger-sdk-android/blob/master/sample/src/com/esri/android/geotrigger/sample/GeotriggerActivity.java
[sample-app-geotrigger-helper]: https://github.com/Esri/geotrigger-sdk-android/blob/master/sample/src/com/esri/android/geotrigger/sample/GeotriggerHelper.java
[goog-dev-console]: https://cloud.google.com/console


CHANGES.md
# Change Log

## v1.2.0 8/05/2015

* NOTE: As of this version, Android API levels <15 are no longer supported.
* Added offline location processing functionality along with local push notifications, enabled with a new function `GeotriggerService.setOfflineProcessingEnabled`.
* Added `"com.esri.android.geotrigger.intent.OFFLINE_RECEIVE"` intent action for receiving local notifications in offline processing mode.
* Added functions `setTriggerDownloadInterval` and  `setTriggerReportUploadInterval` to `GeotriggerService` for managing trigger downloads and trigger event uploads in offline mode.
* Added functions `requestOnDemandTriggerDownload` and `onDemandTriggerReportUpload` to `GeotriggerService` for control over when new and/or updated triggers are downloaded and the report of triggers fired are uploaded in offline mode.
* Added additional interfaces to `GeotriggerBroadcastReceiver` to listen for offline processing events.
* Additional information about offline processing can be found [here][offline-triggers-doc].

## v1.1.0 2/02/2015

* NOTE: As of this version, Google Play Services is no longer optional and is now a requirement.
* Improved support for Android Studio and Gradle build system. Eclipse is no longer officially supported.
* Updates to support the latest Google Play Services version.
* Added `setServiceStartSticky` method to the `GeotriggerService`, which may help keep the service running longer on some devices which are older or memory constrained. This will also enable tracking to continue after the user has swiped the app out of the recent apps list. If this option is enabled, it is strongly recommended that you add a clear way for the user to stop tracking.

## v1.0.2 5/30/2014

* Added `GeotriggerService.setSslCertPinningEnabled()` to allow control over the use of SSL certificate pinning for requests made by the SDK to the Geotrigger API. The default value is `false`, meaning that SSL certificate pinning will not be used. All requests are made using SSL encryption regardless of this setting.

## v1.0.1 3/19/2014

* Slighty tweaks to some internal behavior. Should not result in observable differences.
* Fixed a bug where rapidly switching in and out of Adaptive would flush Geofences improperly.

## v1.0.0 2/18/2014

* Renamed `GeotriggerBroadcastReceiver.DeviceReadyListener` to `GeotriggerBroadcastReceiver.ReadyListener`, and renamed `onDeviceReady()` to `onReady()`.
* Improved robustness of GCM registration.
* Improvements to ADAPTIVE and ROUGH profiles.
* Miscellaneous bug fixes.

## v0.1.2 1/23/2014:

* The SDK will now throw a RuntimeException if Google Play Services is not included as a dependency, unless `setLegacyTrackingModeFallbackEnabled()` is called with `true` as the second parameter value.
* The intents used for receiving events from the SDK have had their respective actions slightly renamed. If you were registering your own subclass of `GeotriggerBroadcastReceiver` in your `AndroidManiest.xml` to receive intents sent by the SDK, the action names within your `<intent-filter>` will need to be adjusted to match what is shown [here][handling-events-doc]. (Actions that started with `com.esri.android.action` now start with `com.esri.android.geotrigger.action`).
* Reduced the number of network requests necessary when oauth tokens expires.
* HTTP requests made through the `GeotriggerApiClient` class methods are now sequential.
* The static method `GeotriggerService.hardStop()` has been renamed to `GeotriggerService.stop()`.

## v0.1.1 1/14/2014:

* Clarified log messages.
* Clarified some javadoc around resource paths for sounds and icons.
* Changed log tag to include version info.
* Added function `GeotriggerService.getTrackingProfile()` to return the tracking profile that was most recently used.

[handling-events-doc]: https://developers.arcgis.com/en/geotrigger-service/guide/android-handling-events/
[offline-triggers-doc]: https://developers.arcgis.com/en/geotrigger-service/guide/android-offline-triggers/


javadoc/reference/lists.js
var DATA = [
      { id:0, label:"com.esri.android.geotrigger", link:"com/esri/android/geotrigger/package-summary.html", type:"package" },
      { id:1, label:"com.esri.android.geotrigger.GeotriggerApiClient", link:"com/esri/android/geotrigger/GeotriggerApiClient.html", type:"class" },
      { id:2, label:"com.esri.android.geotrigger.GeotriggerApiListener", link:"com/esri/android/geotrigger/GeotriggerApiListener.html", type:"class" },
      { id:3, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.html", type:"class" },
      { id:4, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.DeviceRegisterFailListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.DeviceRegisterFailListener.html", type:"class" },
      { id:5, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.GeofenceCreationListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.GeofenceCreationListener.html", type:"class" },
      { id:6, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.GeofenceRemovalListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.GeofenceRemovalListener.html", type:"class" },
      { id:7, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.LocationUpdateListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.LocationUpdateListener.html", type:"class" },
      { id:8, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.LocationUploadFailListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.LocationUploadFailListener.html", type:"class" },
      { id:9, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.LocationUploadListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.LocationUploadListener.html", type:"class" },
      { id:10, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.ProcessLocationFailListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.ProcessLocationFailListener.html", type:"class" },
      { id:11, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.ProcessLocationListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.ProcessLocationListener.html", type:"class" },
      { id:12, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.PushMessageListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.PushMessageListener.html", type:"class" },
      { id:13, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.PushTokenRegisterFailListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.PushTokenRegisterFailListener.html", type:"class" },
      { id:14, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.PushTokenRegisterListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.PushTokenRegisterListener.html", type:"class" },
      { id:15, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.ReadyListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.ReadyListener.html", type:"class" },
      { id:16, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.TrackingProfileChangeListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.TrackingProfileChangeListener.html", type:"class" },
      { id:17, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.TriggerDownloadFailListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.TriggerDownloadFailListener.html", type:"class" },
      { id:18, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.TriggerDownloadListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.TriggerDownloadListener.html", type:"class" },
      { id:19, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.TriggerReportUploadFailListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.TriggerReportUploadFailListener.html", type:"class" },
      { id:20, label:"com.esri.android.geotrigger.GeotriggerBroadcastReceiver.TriggerReportUploadListener", link:"com/esri/android/geotrigger/GeotriggerBroadcastReceiver.TriggerReportUploadListener.html", type:"class" },
      { id:21, label:"com.esri.android.geotrigger.GeotriggerService", link:"com/esri/android/geotrigger/GeotriggerService.html", type:"class" },
      { id:22, label:"com.esri.android.geotrigger.MessageReceiver", link:"com/esri/android/geotrigger/MessageReceiver.html", type:"class" },
      { id:23, label:"com.esri.android.geotrigger.TriggerBuilder", link:"com/esri/android/geotrigger/TriggerBuilder.html", type:"class" }

    ];
